src/server/mod.rs

```diff
@@ -176,13 +176,24 @@ where S: NetworkStream + Clone, H: Handler {
 /// A listening server, which can later be closed.
 pub struct Listening {
     #[cfg(feature = "nightly")]
-    _guard: JoinHandle<()>,
+    _guard: Option<JoinHandle<()>>,
     #[cfg(not(feature = "nightly"))]
-    _guard: JoinHandle,
+    _guard: Option<JoinHandle>,
     /// The socket addresses that the server is bound to.
     pub socket: SocketAddr,
 }
 
+impl Drop for Listening {
+    fn drop(&mut self) {
+        match self._guard.take() {
+            Some(guard) => {
+                let _ = guard.join();
+            },
+            None => ()
+        }
+    }
+}
+
 impl Listening {
     /// Stop the server from listening to its socket address.
     pub fn close(&mut self) -> HttpResult<()> {
```

Comment:
This function should probably take the guard for now, so that it at least lets you close a program running a server.